packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -287,6 +300,38 @@ export const withBlockBindingSupport = createHigherOrderComponent(
 			]
 		);
 
+		// Throw a warning if the block is connected to an invalid source.
+		if ( invalidBinding ) {
+			const removeAllBindingsButton = (
+				<Button
+					__next40pxDefaultSize={ false }
+					key="remove-all-bindings"
+					onClick={ () =>
+						updateBlockBindings( {
+							[ invalidBinding.attribute ]: undefined,
+						} )
+					}
+					variant="primary"
+				>
+					{ __( 'Remove connection' ) }
+				</Button>
```

Comment:
Do you think it is worth adding the button to remove the invalid binding?

Reply:
I should wait for this option for more feedback. This case should only happen on development sites, so maybe it does not worth the extra work.

Removing connections with the UI requires some knowledge about how tools component work. And I guess people will ask for a button to add or remove connections via block ToolsPanel (the same you use for putting the text bold).

Reply:
You're right that if we only allow admin users to create and modify bindings, it could be "risky" to add this button for all users. Maybe we can just remove it and add it if considered necessary in the future. What do you think?

Reply:
> I should wait for this option for more feedback. This case should only happen on development sites, so maybe it does not worth the extra work.

I'm not sure this is correct. What about the case wherein a plugin adds a block bindings source, the source is used in a block, and the plugin is later uninstalled? I believe this warning would appear in those cases, though I could be wrong.

> You're right that if we only allow admin users to create and modify bindings, it could be "risky" to add this button for all users. Maybe we can just remove it and add it if considered necessary in the future. What do you think?

I agree that allowing non-admin users to remove bindings could be risky. I also think that omitting the button may introduce inconsistencies in the user experience. Perhaps we should allow design to weigh in?

In any case, I think we should have *some* mechanism for fixing or removing the invalid binding. Notice how, when the connection is broken, we have no block settings — and, consequently, no attributes panel — in the sidebar:

<img width="1143" alt="Screenshot 2024-09-01 at 4 14 34 PM" src="https://github.com/user-attachments/assets/43de8e92-a385-4ff4-b95d-c470bada213c">

Let's say you install a plugin with a custom binding source, make a few connections, but later decide you no longer want to use the plugin. How are admins supposed to not just be alerted to the broken bindings, but given a way to resolve them? 

If we do remove the button for the non-admin users, we could give them a slightly different warning. Maybe something like, "Attribute "content" is connected to undefined "core/undefined" block bindings source. You have insufficient permissions to modify it. Please contact your site administrator."

Reply:
> What about the case wherein a plugin adds a block bindings source, the source is used in a block, and the plugin is later uninstalled? I believe this warning would appear in those cases, though I could be wrong.

Yes, if a source is unregistered (by uninstalling the plugin), the warning will show. It works the same way with unregistered blocks.

I don't know if showing different messages for different users is something done in other places. Personally, I would lean towards:
* If the user can edit the post, show the button even if it is not an admin. I think removing broken connections is not that risky.
* Don't show the button until it is clearer how it should work.